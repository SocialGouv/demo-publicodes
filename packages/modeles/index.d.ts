// Currenty we systematically bundle all the rules even if we only need a
// sub-section of them. We might support "code-splitting" the rules in the
// future.
import { Rule } from "publicodes";

export type { Names as OrientationCovidNames } from "./dist/orientation-covid/index.d.ts";

//export type DottedName = Names;

declare let rules: Record<key, Rule>;

declare let models: Record<key, rules>;

export default models;
