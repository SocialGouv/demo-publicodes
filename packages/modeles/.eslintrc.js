module.exports = {
  extends: ["prettier", "eslint:recommended"],
};
