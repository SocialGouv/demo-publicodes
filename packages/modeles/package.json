{
  "name": "@socialgouv/publicodes-demo-modeles",
  "version": "1.0.0",
  "description": "Des règles publicodes de démo",
  "type": "module",
  "main": "./dist/index.js",
  "module": "dist/index.js",
  "types": "./index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/socialgouv/demo-publicodes.git",
    "directory": "packages/modeles"
  },
  "homepage": "https://github.com/socialgouv/demo-publicodes",
  "license": "MIT",
  "files": [
    "dist",
    "index.d.ts"
  ],
  "devDependencies": {
    "eslint-config-prettier": "^8.5.0",
    "jest": "^29.3.1",
    "js-yaml": "^4.1.0",
    "onchange": "^7.1.0",
    "publicodes": "^1.0.0-beta.55"
  },
  "peerDependencies": {
    "publicodes": "^1.0.0-beta.55"
  },
  "scripts": {
    "build": "node ./scripts/build-rules.js",
    "start": "yarn run build && onchange 'règles/**/*.yaml' -- yarn run build",
    "clean": "rimraf dist node_modules",
    "prepack": "yarn run build",
    "test": "yarn node --experimental-vm-modules $(yarn bin jest)"
  },
  "dependencies": {
    "camelcase": "^7.0.0"
  }
}
