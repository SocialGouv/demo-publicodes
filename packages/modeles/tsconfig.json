{
	"extends": "tsconfig/base.json",
	"compilerOptions": {
		"allowJs": true,
		"noEmit": true,
		"strict": true
	},
	"include": ["index.d.ts", "dist/**/*.ts"]
}
