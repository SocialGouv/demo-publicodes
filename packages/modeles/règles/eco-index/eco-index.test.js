/* eslint-env node18 */
import yaml from "js-yaml";
import Engine, { formatValue } from "publicodes";

import modeles from "@socialgouv/publicodes-demo-modeles";

const rules = yaml.load(modeles.ecoIndex.yaml);

test("a besoin de variables pour fonctionner", () => {
  const engine = new Engine(rules);
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.missingVariables).toMatchSnapshot();
});

const defaultSituation = {};

test("ne renvoie rien par défaut", () => {
  const engine = new Engine(rules);
  engine.setSituation(defaultSituation);
  const evaluated = engine.evaluate("résultat");
  expect(evaluated.nodeValue).toMatchSnapshot();
  expect(evaluated.missingVariables).toMatchSnapshot();
});

const tests = [
  {
    title: "100, 200, 300",
    situation: {
      "input . dom": 100,
      "input . req": 200,
      "input . size": 300,
    },
    expected: {
      ecoindex: "62",
      "ecoindex . grade": "C",
      "emissions gaz": "1,76\u00A0gCO2e",
      "consommation eau": "2,64\u00A0cl",
      "emissions . km": "19,6\u00A0km",
      "emissions . douches": "0,4\u00A0douche(s)",
    },
  },
  {
    title: "1000, 2000, 3000",
    situation: {
      "input . dom": 1000,
      "input . req": 2000,
      "input . size": 3000,
    },
    expected: {
      ecoindex: "19",
      "ecoindex . grade": "F",
      "emissions gaz": "2,62\u00A0gCO2e",
      "consommation eau": "3,93\u00A0cl",
      "emissions . km": "29,1\u00A0km",
      "emissions . douches": "0,7\u00A0douche(s)",
    },
  },
];

test.each(tests)("$title", ({ title, situation, expected }) => {
  const engine = new Engine(rules);
  engine.setSituation({
    ...defaultSituation,
    ...situation,
  });
  Object.keys(expected).forEach((key) => {
    const evaluated = engine.evaluate(key);
    expect(`${key}: ${formatValue(evaluated)}`).toEqual(
      `${key}: ${expected[key]}`
    );
  });
});
