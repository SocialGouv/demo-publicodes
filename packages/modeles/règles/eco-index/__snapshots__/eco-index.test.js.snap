// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a besoin de variables pour fonctionner 1`] = `
{
  "input . dom": 2,
  "input . req": 2,
  "input . size": 2,
}
`;

exports[`ne renvoie rien par défaut 1`] = `undefined`;

exports[`ne renvoie rien par défaut 2`] = `
{
  "input . dom": 2,
  "input . req": 2,
  "input . size": 2,
}
`;
